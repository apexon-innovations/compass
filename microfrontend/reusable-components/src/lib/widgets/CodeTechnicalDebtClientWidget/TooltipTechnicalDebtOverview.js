import React from 'react'
import PropTypes from 'prop-types'

// Style
import style from './CodeTechnicalDebtClientWidget.module.scss'

const TooltipTechnicalDebtOverview = props => {
  const { active, payload } = props
  if (active) {
    const data = payload[0].payload
    return (
      <div className={['graphToolTip', style.ttTechnicalDebtOverview].join(' ')}>
        <ul className={style.tooltipHead}>
          <li>{data.name}</li>
          <li>
            Total - <span className={style.totalCount}>{data.totalDays} days</span>
          </li>
        </ul>
        <ul className={style.tooltipBody}>
          {data.maintainability ? (
            <li className={style.mainatainability}>
              Maintainability - <span>{data.maintainability} days</span>
            </li>
          ) : (
            ''
          )}
          {data.reliability ? (
            <li className={style.reliability}>
              Reliability - <span>{data.reliability} days</span>
            </li>
          ) : (
            ''
          )}
          {data.security ? (
            <li className={style.security}>
              Security - <span>{data.security} days</span>
            </li>
          ) : (
            ''
          )}
        </ul>
      </div>
    )
  }
  return null
}

TooltipTechnicalDebtOverview.propTypes = {
  active: PropTypes.any,
  payload: PropTypes.any,
}

export default TooltipTechnicalDebtOverview
