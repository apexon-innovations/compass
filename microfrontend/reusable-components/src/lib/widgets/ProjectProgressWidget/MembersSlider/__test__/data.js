const data = {
  data: [
    {
      email: 'Pradeep.shrivastava@infostretch.com',
      name: 'Pradeep Shrivastava',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933',
    },
    {
      email: 'vijay.solanki@infostretch.com',
      name: 'Vijay Solanki',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621',
    },
    {
      email: 'suraj.bhosale@infostretch.com',
      name: 'Suraj Bhosale',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950',
    },
    {
      email: 'yatin.bhavsar@infostretch.com',
      name: 'Yatin Bhavsar',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973',
    },
    {
      email: 'vishal.kumar@infostretch.com',
      name: 'Vishal kumar',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930',
    },
    {
      email: 'gunjan.shah@infostretch.com',
      name: 'Gunjan Shah',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
    },
    {
      email: 'namra.porecha@infostretch.com',
      name: 'Namra Porecha',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375',
    },
    {
      email: 'purushottam.Khandebharad@infostretch.com',
      name: 'Purushottam Khandebharad',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437',
    },
    {
      email: 'amit.ghadge@infostretch.com',
      name: 'Amit Ghadge',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269',
    },
  ],
}

export default data
