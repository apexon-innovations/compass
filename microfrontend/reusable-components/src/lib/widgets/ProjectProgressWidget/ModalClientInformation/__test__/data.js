const data = {
  data: {
    id: '5ee31002d48d514e801cb10b',
    name: 'Compass',
    clientName: 'Infostretch',
    initials: 'C',
    category: 'All (Dev & QA & Support)',
    iconLocation:
      'http://sampleicon.com/image.png',
    billingType: 'Internal',
    startDate: 'Wed Jan 01 00:00:00 UTC 2020',
    endDate: 'Thu Jan 02 00:00:00 UTC 2020',
    resources: {
      actual: 10,
      sow: 10,
    },
    healthMetrics: {
      overall: 'R',
      periodicReview: 'G',
      scope: 'A',
      deliverables: 'A',
      customer: 'G',
      schedule: 'R',
      financials: 'G',
      people: 'G',
      compliance: 'G',
      startUp: 'G',
      process: 'G',
      planning: 'G',
    },
    projectManager: [],
    deliveryManager: [
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
        nestProjects: ['Infostretch - DES'],
      },
    ],
    accountManager: [
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
        nestProjects: ['Infostretch - DES'],
      },
    ],
    teamMembers: [
      {
        email: 'Pradeep.shrivastava@infostretch.com',
        name: 'Pradeep Shrivastava',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933',
      },
      {
        email: 'vijay.solanki@infostretch.com',
        name: 'Vijay Solanki',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621',
      },
      {
        email: 'suraj.bhosale@infostretch.com',
        name: 'Suraj Bhosale',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950',
      },
      {
        email: 'yatin.bhavsar@infostretch.com',
        name: 'Yatin Bhavsar',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973',
      },
      {
        email: 'vishal.kumar@infostretch.com',
        name: 'Vishal kumar',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930',
      },
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
      },
      {
        email: 'namra.porecha@infostretch.com',
        name: 'Namra Porecha',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375',
      },
      {
        email: 'purushottam.Khandebharad@infostretch.com',
        name: 'Purushottam Khandebharad',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437',
      },
      {
        email: 'amit.ghadge@infostretch.com',
        name: 'Amit Ghadge',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269',
      },
    ],
  },
  fileUploaderUrl:
    'https://compass-dev-api.infostretch.com/psr-service/project/5ee31002d48d514e801cb10b/logo',
}

export default data
