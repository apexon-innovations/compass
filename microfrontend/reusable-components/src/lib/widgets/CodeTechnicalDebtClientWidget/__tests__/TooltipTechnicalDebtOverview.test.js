/* eslint-disable */
import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { sleep } from '../../../../__tests__/commonTestFunction'
import TooltipTechnicalDebtOverview from '../TooltipTechnicalDebtOverview'
import { toolTipData } from './data'

const props = { ...toolTipData }

describe(`TooltipTechnicalDebtOverview`, () => {
  let container = null

  beforeEach(() => {
    // set up a DOM element as a render target
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container)
    container.remove()
    container = null
  })

  it(`Snapshot`, async () => {
    act(() => {
      render(<TooltipTechnicalDebtOverview {...props} />, container)
    })
    await act(() => sleep(500))
    expect(container).toMatchSnapshot()
  })

  it(`Snapshot No data`, async () => {
    act(() => {
      render(<TooltipTechnicalDebtOverview />, container)
    })
    await act(() => sleep(500))
    expect(container).toMatchSnapshot()
  })
})
