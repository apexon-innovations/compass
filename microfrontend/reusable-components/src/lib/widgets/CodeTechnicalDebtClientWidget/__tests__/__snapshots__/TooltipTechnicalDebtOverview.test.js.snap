// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipTechnicalDebtOverview Snapshot 1`] = `
<div>
  <div
    class="graphToolTip "
  >
    <ul>
      <li>
        Jan-2021
      </li>
      <li>
        Total - 
        <span>
          1.4
           days
        </span>
      </li>
    </ul>
    <ul>
      <li>
        Maintainability - 
        <span>
          1.4
           days
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`TooltipTechnicalDebtOverview Snapshot No data 1`] = `<div />`;
